### ISSUE TITLE
Exclude cluster-managed endpoints from restores

### ISSUE THREAD

**Issue by ncdc** [heptio/ark#72]
It doesn't make sense to restore endpoints that are managed by the endpoints controller.

**Comment by jrnt30** [comment]
tl;dr - Do we even need this?  As I am playing with the different permutations of this, I am observing that: 

- If we restore a set of Endpoints that no longer match nodes/pods (as in a restore) they will only exist until a service with the same name as the set of endpoints 
- As soon as a there are matching pods for the given service it "takes over" that endpoint object completely and wipes out any "stagnant" endpoints that may 
- In cases where the there does not exist a service with the same name/uid (see below) as the endpoints in the proposed solution we would be treating these as "user defined" endpoints anyways, unless you have a better way of solidifying the relationship between a "service managed endpoints" and "user managed endpoints"

## Potential solution if needed

In followup to #99 I'm unclear at this point how I could determine on the `v1.Endpoints` resource which of those are created automatically via the Service's controller or manually.  Looking at the domain, I do not specifically see any reference or attribution at the Endpoint level back to the Service that created it.  

One cross-resource heuristic that we may be able to use would be: 
- Collect all of the `Service[*].metadata.name` (it looks like we could also link via `Service[*].metadata.uid` but that seems more opaque)
- If the `endpoint.metadata.name` is included in the the Service Name list, omit the explicit creation of it. 
- Else we assume that it is a "manual" endpoint creation and create it. 

This would be somewhat complicated if you had a backup or restore that excluded the `services` resource but not `endpoints`.  In this case, we would create those endpoints as they would appear "manually" created, which may be fine.
